Source/DafnyRuntime/DafnyRuntimePython/_dafny/__init__.py

```diff
@@ -177,8 +254,11 @@ def __add__(self, other):
 
     def __getitem__(self, key):
         if isinstance(key, slice):
-            indices = range(*key.indices(len(self)))
-            return Seq((self.Elements[i] for i in indices), isStr=self.isStr)
+            start, stop, step = key.indices(len(self))
+            if isinstance(self.elems, _SeqSlice):
+                # Avoiding .Elements call on SeqSlice avoids creating a list, which is expensive
+                return Seq(_SeqSlice(self.elems, start=start, stop=stop, step=step), isStr=self.isStr)
+            return Seq(_SeqSlice(self.Elements, start=start, stop=stop, step=step), isStr=self.isStr)
```

Comment:
```suggestion
            elements = self.elems if isinstance(self.elems, _SeqSlice) else self.Elements
            return Seq(_SeqSlice(elements, start=start, stop=stop, step=step), isStr=self.isStr)
```
Something along these lines?